test/distributed/tensor/test_redistribute.py

```diff
@@ -534,10 +544,11 @@ def test_redistribute_shard_dim_change(self, dtype):
                         1,
                     )
                 else:
-                    self.assertEqual(
-                        comm_mode.get_comm_counts()[funcol.all_gather_into_tensor],
-                        1,
-                    )
+                    if not self.is_local_tensor_enabled:
+                        self.assertEqual(
+                            comm_mode.get_comm_counts()[funcol.all_gather_into_tensor],
```

Comment:
arguably LocalTensor should update the comm counts correctly; another good task to farm out